### PR TITLE
Mars Protocol Perps Volume and Revenue

### PR DESCRIPTION
**Pull request by linkielink**
This PR adds 24h Volume and 24h Revenue to the Mars Protocol Perps (`mars-perp`) implementation

### CODE REVIEW

**Comment by llamabutler** [comment]
The mars-perp adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs mars-perp
    
    🦙 Running MARS-PERP adapter 🦙
    ---------------------------------------------------
    Start Date:	Fri, 13 Dec 2024 08:28:35 GMT
    End Date:	Sat, 14 Dec 2024 08:28:35 GMT
    ---------------------------------------------------
    
    Version -> DERIVATIVES
    ---------
    NEUTRON 👇
    Backfill start time: 13/12/2024
    Daily volume: 90.41 k
    Daily revenue: 91
    End timestamp: 1734164914 (2024-12-14T08:28:34.000Z)

File: dexs/mars-perp/index.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    last24HourFees = last24HourTradingFee + last24HourFundingFee;
+
+    fetchTimestamp = Math.round(
+      new Date(globalOverview.daily_trading_volume[1].date).getTime() / 1000
+    );
+  }
+
+  return {
+    dailyVolume: last24HourVolume,
+    dailyRevenue: last24HourFees,
```
Comment (dtmkeng) [inline dexs/mars-perp/index.ts]: fees it require daily fees field `dailyFees` could you add daily fees

**Review by dtmkeng — CHANGES_REQUESTED** [review]

**Comment by llamabutler** [comment]
The mars-perp adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs mars-perp
    
    🦙 Running MARS-PERP adapter 🦙
    ---------------------------------------------------
    Start Date:	Sat, 14 Dec 2024 05:20:39 GMT
    End Date:	Sun, 15 Dec 2024 05:20:39 GMT
    ---------------------------------------------------
    
    Version -> DERIVATIVES
    ---------
    NEUTRON 👇
    Backfill start time: 13/12/2024
    Daily volume: 36.07 k
    Daily revenue: 32
    Daily fees: 130
    End timestamp: 1734240038 (2024-12-15T05:20:38.000Z)

File: dexs/mars-perp/index.ts (hunk 1 above)
Comment (linkielink) [reply dexs/mars-perp/index.ts]: > fees it require daily fees field `dailyFees` could you add daily fees

I must have mistaken revenue with fees. I updated the logic and added fees and the correct amount of revenue for the protocol.

**Comment by llamabutler** [comment]
The mars-perp adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs mars-perp
    
    🦙 Running MARS-PERP adapter 🦙
    ---------------------------------------------------
    Start Date:	Sat, 14 Dec 2024 11:27:04 GMT
    End Date:	Sun, 15 Dec 2024 11:27:04 GMT
    ---------------------------------------------------
    
    Version -> DERIVATIVES
    ---------
    NEUTRON 👇
    Backfill start time: 13/12/2024
    Daily volume: 36.07 k
    Daily protocol revenue: 32
    Daily fees: 130
    Daily short open interest: 9.13 k
    Daily long open interest: 20.37 k
    Daily open interest: 29.50 k
    End timestamp: 1734262023 (2024-12-15T11:27:03.000Z)

**Comment by llamabutler** [comment]
The mars-perp adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs mars-perp
    
    🦙 Running MARS-PERP adapter 🦙
    ---------------------------------------------------
    Start Date:	Sat, 14 Dec 2024 20:35:08 GMT
    End Date:	Sun, 15 Dec 2024 20:35:08 GMT
    ---------------------------------------------------
    
    2024-12-14
    Version -> DERIVATIVES
    ---------
    NEUTRON 👇
    Backfill start time: 13/12/2024
    Daily volume: 36.07 k
    Daily protocol revenue: 32
    Daily fees: 130
    Daily short open interest: 9.13 k
    Daily long open interest: 20.37 k
    Daily open interest: 29.50 k
    End timestamp: 1734294907 (2024-12-15T20:35:07.000Z)

**Comment by llamabutler** [comment]
The mars-perp adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs mars-perp
    
    🦙 Running MARS-PERP adapter 🦙
    ---------------------------------------------------
    Start Date:	Sat, 14 Dec 2024 20:36:36 GMT
    End Date:	Sun, 15 Dec 2024 20:36:36 GMT
    ---------------------------------------------------
    
    Version -> DERIVATIVES
    ---------
    NEUTRON 👇
    Backfill start time: 13/12/2024
    Daily volume: 36.07 k
    Daily protocol revenue: 32
    Daily fees: 130
    Daily short open interest: 9.13 k
    Daily long open interest: 20.37 k
    Daily open interest: 29.50 k
    End timestamp: 1734294995 (2024-12-15T20:36:35.000Z)

**Comment by linkielink** [comment]
I added `dailyProtocolRevenue`, `dailyFees`, `dailyShortOpenInterest`, `dailyLongOpenInterest`, `dailyOpenInterest` and added backwards compatibility based on `fromApi.timestamp` (I assume that's the timestamp the data is fetched for).

**Comment by llamabutler** [comment]
The mars-perp adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs mars-perp
    
    🦙 Running MARS-PERP adapter 🦙
    ---------------------------------------------------
    Start Date:	Sat, 14 Dec 2024 20:42:26 GMT
    End Date:	Sun, 15 Dec 2024 20:42:26 GMT
    ---------------------------------------------------
    
    Version -> DERIVATIVES
    ---------
    NEUTRON 👇
    Backfill start time: 13/12/2024
    Daily volume: 36.07 k
    Daily protocol revenue: 32
    Daily fees: 130
    Daily short open interest: 9.13 k
    Daily long open interest: 20.37 k
    Daily open interest: 29.50 k
    End timestamp: 1734295345 (2024-12-15T20:42:25.000Z)

**Pull request merged**